common/src/main/java/com/microsoft/identity/common/internal/cache/ADALTokenCacheItem.java

```diff
@@ -73,19 +76,22 @@
         mRawIdToken = response.getIdToken();
         if (account instanceof AzureActiveDirectoryAccount) {
             mUserInfo = new ADALUserInfo((AzureActiveDirectoryAccount) account);
+            mTenantId = ((AzureActiveDirectoryAccount) account).getTenantId();
         }
 
-        /*
         if(accessToken instanceof AzureActiveDirectoryAccessToken){
-            mExpiresOn = ((AzureActiveDirectoryAccessToken)accessToken).getExpiresOn()
+            mExpiresOn = ((AzureActiveDirectoryAccessToken)accessToken).getExpiresOn();
+            mExtendedExpiresOn = ((AzureActiveDirectoryAccessToken)accessToken).getExtendedExpiresOn();
         }
 
         if(refreshToken instanceof AzureActiveDirectoryRefreshToken){
-            mExpiresOn = ((AzureActiveDirectoryTokenResponse)response).getExpiresOn();
+            mIsMultiResourceRefreshToken = ((AzureActiveDirectoryRefreshToken)refreshToken).getIsFamilyRefreshToken();
+            mFamilyClientId = ((AzureActiveDirectoryRefreshToken)refreshToken).getFamilyId();
         }
 
-        */
-
+        if(response instanceof AzureActiveDirectoryTokenResponse){
+            mSpeRing = ((AzureActiveDirectoryTokenResponse)response).getSpeRing();
```

Comment:
instead of doing instanceof, would you consider putting getSpeRing in the base class as abstract and implement it in the  child classes. where AzureActiveDirectoryTokenResponse returns true or false, every other child class always returns false

Reply:
No.  It does not apply to any identity provider other than AAD.  So doesn't belong in the base class.